legend-engine-xt-relationalStore-sqlserver-execution-tests/src/main/java/org/finos/legend/engine/authentication/SqlServerTestDatabaseAuthenticationFlowProvider.java

```diff
@@ -0,0 +1,44 @@
+// Copyright 2021 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.authentication;
+
+import org.eclipse.collections.api.list.ImmutableList;
+import org.eclipse.collections.impl.factory.Lists;
+import org.finos.legend.engine.authentication.flows.SqlServerStaticWithUserPasswordFlow;
+import org.finos.legend.engine.authentication.provider.AbstractDatabaseAuthenticationFlowProvider;
+import org.finos.legend.engine.authentication.provider.DatabaseAuthenticationFlowProviderConfiguration;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.connection.authentication.AuthenticationStrategy;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.relational.connection.specification.DatasourceSpecification;
+
+public class SqlServerTestDatabaseAuthenticationFlowProvider extends AbstractDatabaseAuthenticationFlowProvider
```

Comment:
I don't remember the details now .. but I think we have a bug in the flow provider/selector singleton implementation.  Something like if you have multiple flow providers in the CP, depending on how you initialize them, LegendDefaultDatabaseAuthenticationFlowProvider.configure throws an exception.

Have you run the full test suite with this change ?

Reply:
SqlServerTest flow is only used in this module, and it does not conflicts with LegendDefault flow in any module. What sort of tests you have in mind? (Tc was green)